packages/css/src/rules/index.ts

```diff
@@ -217,6 +220,38 @@ function processCompoundStyle(
     : css(style, getDebugName(debugId, `compound_${index}`));
 }
 
+export function rulesVariants<Slots extends SlotRecipeDefinition>(
+  slots: Slots,
+  debugId?: string
+): SlotRecipeResult<Slots> {
+  const result: Partial<SlotRecipeResult<Slots>> = {};
+
+  Object.entries(slots).forEach(([slotName, slotConfig]) => {
+    const slotDebugId = debugId ? `${debugId}_${slotName}` : slotName;
+    result[slotName as keyof Slots] = rules(
+      {
+        base: slotConfig.base,
+        variants: slotConfig.variants,
+        toggles: slotConfig.toggles,
+        ...slotConfig
+      } as PatternOptions<
+        Slots[typeof slotName]["variants"],
+        Slots[typeof slotName]["toggles"],
+        undefined
+      >,
+      slotDebugId
+    ) as RuntimeFn<
+      ConditionalVariants<
+        Slots[typeof slotName]["variants"],
+        Slots[typeof slotName]["toggles"]
+      >,
+      ComplexPropDefinitions<PropTarget>
+    >;
+  });
+
+  return result as SlotRecipeResult<Slots>;
+}
```

Comment:
There should be a Mapping feature to maintain the [same usability as cssVariant](](https://github.com/mincho-js/mincho/blob/871e38ae97311961dfd3adc543279459a70c5f68/packages/css/src/css/index.ts#L148)).